src/Enhavo/Bundle/NewsletterBundle/Controller/PendingSubscriberController.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Enhavo\Bundle\NewsletterBundle\Controller;
+
+use Enhavo\Bundle\NewsletterBundle\Pending\PendingSubscriberManager;
+use Enhavo\Bundle\NewsletterBundle\Subscription\SubscriptionManager;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class PendingSubscriberController extends AbstractController
+{
+    /** @var SubscriptionManager */
+    private $subscriptionManager;
+
+    /** @var PendingSubscriberManager */
+    private $pendingManager;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /**
+     * SubscriptionController constructor.
+     * @param SubscriptionManager $subscriptionManager
+     * @param PendingSubscriberManager $pendingManager
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(SubscriptionManager $subscriptionManager, PendingSubscriberManager $pendingManager, TranslatorInterface $translator)
+    {
+        $this->subscriptionManager = $subscriptionManager;
+        $this->pendingManager = $pendingManager;
+        $this->translator = $translator;
+    }
+
+    public function activateAction(Request $request)
+    {
+        $id = $request->get('id');
```

Comment:
For security reasons, we should use a token here instead of the id. Users could easily activate other users

Reply:
enhavo_newsletter_pending_subscriber_activate (admin): 
this is a admin only route. you have to be logged in as admin/superadmin to call it

enhavo_newsletter_subscribe_activate (theme):
SubscriptionController:activateAction uses token

Reply:
Ah ok i see